base/Plain_netCDF_Time.F90

```diff
@@ -482,18 +485,70 @@ subroutine parse_timeunit_i8(tunit, n, t0, dt, rc)
 !    write(6,*) 'y, c1, m, c1, d',  y, c1, m, c1, d
 !    write(6,*) 'hour, c1, min, c1, sec', hour, c1, min, c1, sec
 
-    _ASSERT(trim(s_unit) == 'seconds', "s_unit /= 'seconds' is not handled")
-    isec=n
+    if (trim(s_unit) == 'seconds') then
+       isec=n
+    elseif (trim(s_unit) == 'minutes') then
+       isec=n * 60
+    elseif (trim(s_unit) == 'hours') then
+       isec=n * 3600
+    else
+       _FAIL ('time_unit not implemented')
+    end if
 
-    gregorianCalendar = ESMF_CalendarCreate(ESMF_CALKIND_GREGORIAN, name='Gregorian_obs', _RC)
-    call ESMF_timeSet(t0, yy=y,mm=m,dd=d,h=hour,m=min,s=sec,&
-         calendar=gregorianCalendar, _RC)
+    call ESMF_timeSet(t0, yy=y,mm=m,dd=d,h=hour,m=min,s=sec, _RC)
     call ESMF_timeintervalSet(dt, d=0, h=0, m=0, s_i8=isec, _RC)
 
     _RETURN(_SUCCESS)
 
   end subroutine parse_timeunit_i8
 
+  subroutine diff_two_timeunits (tunit1, tunit2, x, rc)
+    character(len=*), intent(in) :: tunit1
```

Comment:
Wouldn't this be easier to use and understand if it was a function that returns x rather than having an `INTENT(OUT)` argument ?

Reply:
I realized there a need to add an optional output dt_ESMF to the subroutine.